File: 01-week-1--starter-algorithms/03-day-4--selection-sort/ruby/selection_sort.rb
def selection_sort(arr)
  sortArray = []
  until arr.length == 0
    minimum = arr.min
    index = arr.index(minimum)
    sortArray << minimum
    arr.delete_at(index)
  end
  sortArray
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort([3, -1, 5, 2])

  puts

  puts "Expecting: [-7, 7, 77, 777]"
  print "=> "
  print selection_sort([77, 777, 7, -7])

  # Don't forget to add your own!

  # BENCHMARK HERE, you can print the average runtime
  long_input = []

  100.times { long_input << rand }

  start_time = Time.now
  1000.times do
    selection_sort([2, 1])
    selection_sort(long_input)
  end
  avg_time = (Time.now - start_time) / 2000
  puts avg_time
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 01-week-1--starter-algorithms/03-day-4--selection-sort/javascript/selection_sort.js
function selectionSort(arr) {
  const sortArray = []
  while (arr.length > 0){
    let minimun = Math.min(...arr)
    let index = arr.indexOf(minimun)
    sortArray.push(minimun)
    arr.splice(index, 1)
  }
  return sortArray
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSort([3, -1, 5, 2]));

  console.log("");

  console.log("Expecting: [-7, 7, 77, 777]");
  console.log("=>", selectionSort([77, 777, 7, -7]));

  // BENCHMARK HERE, and print the average runtime
  const longInput = [];

  for (let i = 0; i < 100; ++i) {
    longInput.push(Math.random());
  }
  
  const startTime = Date.now()
  for (let i = 0; i < 1000; ++i){
    selectionSort([2, 1])
    selectionSort(longInput)
  }
  const avgTime = (Date.now() - startTime) / 2000
  console.log(avgTime)
}

module.exports = selectionSort;

// Please add your pseudocode to this file
// make empty array for sorted result, get the minimums out of input array in to sorted array in order
// And a written explanation of your solution
